query-node/schemas/membership.graphql

```diff
@@ -10,6 +10,26 @@ type AvatarUri @variant {
 
 union Avatar = AvatarObject | AvatarUri
 
+enum MembershipExternalResourceType {
+  EMAIL
+  HYPERLINK
+  TWITTER
+  TELEGRAM
+  DISCORD
+  FACEBOOK
+  YOUTUBE
+  MATRIX
+  IRC
+  WECHAT
+  WHATSAPP
+}
+
+type MembershipExternalResource @entity {
+  type: MembershipExternalResourceType!
+  value: String!
+  memberMetadata: MemberMetadata!
+}
+
```

Comment:
This schema definition looks fine. However, I think if we had custom `ID` input of some unique definition such as concatenation of `memberId` & `resourceType`, then we won't have to deal with comprehensive checks to avoid duplicate records. With some unique ID space, Postgres would ensure no duplicate records (i.e. a member with multiple records of the same type).
e.g.
```graphql
type MembershipExternalResource @entity {
 "Unique combination of {memberId}-{resourceType}"
  id: ID!
  type: MembershipExternalResourceType!
  value: String!
  memberMetadata: MemberMetadata!
}
```

Reply:
Good idea thanks. I implemented this but with the `MemberMetadata.Id` instead of `memberId` because for a given member multiple events refers to different `MemberMetadata` with potentially different external resources (as well as different values for other fields too)